RealmTasks Apple/RealmTasks macOS/ListViewController.swift

```diff
@@ -346,177 +489,152 @@ extension TaskListViewController: NSTableViewDelegate {
     private func updateColors() {
         tableView.enumerateAvailableRowViewsUsingBlock { rowView, row in
             // For some reason tableView.viewAtColumn:row: returns nil while animating, will use view hierarchy instead
-            if let cellView = rowView.subviews.first as? TaskCellView {
-                NSView.animateWithDuration(0.5, animations: {
+            if let cellView = rowView.subviews.first as? ItemCellView {
+                NSView.animate() {
                     cellView.backgroundColor = self.colorForRow(row)
-                })
+                }
             }
         }
     }
 
     private func colorForRow(row: Int) -> NSColor {
-        let fraction = Double(row) / Double(max(13, tableView.numberOfRows))
-        return NSColor.taskColors().gradientColorAtFraction(fraction)
-    }
+        let colors = ItemType.self is Task.Type ? NSColor.taskColors() : NSColor.listColors()
+        let fraction = Double(row) / Double(max(13, list.items.count))
 
-}
-
-// MARK: TaskCellViewDelegate
+        return colors.gradientColorAtFraction(fraction)
+    }
 
-extension TaskListViewController: TaskCellViewDelegate {
+    // MARK: ItemCellViewDelegate
 
-    func cellView(view: TaskCellView, didComplete complete: Bool) {
-        guard let (item, index) = findItemForCellView(view) else {
+    func cellView(view: ItemCellView, didComplete complete: Bool) {
+        guard let (tmpItem, index) = findItemForCellView(view) else {
             return
         }
 
+        // FIXME: workaround for tuple mutability
```

Comment:
Not sure what there is to fix here. Kind of how Swift works. Could do this if you really want:

```swift
guard let itemAndIndex = findItemForCellView(view) else {
    return
}
var item = itemAndIndex.0
let index = itemAndIndex.1
```